src/EmbedTypeManager.php

```diff
@@ -36,4 +36,22 @@ public function __construct(\Traversable $namespaces, CacheBackendInterface $cac
     $this->setCacheBackend($cache_backend, 'embed_type_plugins');
   }
 
+  /**
+   * Determines the Plugins that satisfy a particular category.
+   *
+   * @param string $category
+   *   A string of type category.
+   *
+   * @return array
+   *   An array of plugin definitions.
+   */
+  public function getDefinitionForCategory($category){
+    $definitions = $this->getDefinitions();
+    for($i = 0; $i < count($definitions); $i++){
+      if($definitions[$i]->category == $category){
+        array_push($definitions_for_category,$definitions[$i]);
```

Comment:
The $definitions_for_category variable is still undefined at this point. It needs to be initialized to an empty array before the for loop. Also, we should just be using a `foreach ($defintions as $definition)` instead of a `for` loop.